app/routes/metadata.rb

```diff
@@ -1,28 +1,30 @@
+require_relative '../../x'
+
 module ExercismWeb
   module Routes
     class Metadata < Core
-      get '/exercises/:track_id/:slug' do |track_id, slug|
-        status, response = Xapi.get("v2", "exercises", track_id, slug)
-        data = JSON.parse(response)
-        if status == 404
-          flash[:notice] = data['error']
+      get '/exercises/:id/:slug/tests' do |track_id, slug|
```

Comment:
Since the resource is named `exercises` we can't call the next segment `:id` -- that suggests that it's the `exercises.id`, not the track ID.